### PR TITLE
Add yamllint to CI

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Add yamllint to CI

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -163,9 +164,8 @@ sumologic:
       default:
         name: logs
         config-name: endpoint-logs
-        
         ## Properties can be used to extend default settings, such as processing rules, fields etc
-        #properties:
+        # properties:
         #  filters:
         #    - name: "Test Exclude Debug"
         #      filter_type: "Exclude"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:171]: fix indentation for subkeys

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -538,23 +538,23 @@ fluentd:
...
+
+      # extraVolumes:
+      #   - name: es-certs
+      #     secret:
+      #       defaultMode: 420
+      #       secretName: es-certs
+      # extraVolumeMounts:
+      #   - name: es-certs
+      #     mountPath: /certs
+      #     readOnly: true
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:557]: this is incorrect indentation

Comment (pmalek-sumo) [reply deploy/helm/sumologic/values.yaml]: Sure, my bad.

Just wanted to note that when `comments-indentation` is enabled, `yamllint` complains about those lines (among others)

```
yamllint -c .yamllint.yaml deploy/helm/sumologic/values.yaml
deploy/helm/sumologic/values.yaml
  541:5     error    comment not indented like content  (comments-indentation)
  742:3     error    comment not indented like content  (comments-indentation)
  744:5     error    comment not indented like content  (comments-indentation)
  762:5     error    comment not indented like content  (comments-indentation)
  957:5     error    comment not indented like content  (comments-indentation)
  980:7     error    comment not indented like content  (comments-indentation)
  997:7     error    comment not indented like content  (comments-indentation)
  1010:7    error    comment not indented like content  (comments-indentation)
  1655:5    error    comment not indented like content  (comments-indentation)
  1795:7    error    comment not indented like content  (comments-indentation)
```

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml]: Maybe some newline before comment would help

Comment (pmalek-sumo) [reply deploy/helm/sumologic/values.yaml]: 🤷🏻 
for now I've left it disabled so that we can deal with this later

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml]: sure, please create issue for that if it isn't not created already

Comment (pmalek-sumo) [reply deploy/helm/sumologic/values.yaml]: https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/1178

**Comment by sumo-drosiek** [comment]
One last thing. We should add checks to the vagrant/Makefile as well

**Comment by pmalek-sumo** [comment]
> One last thing. We should add checks to the vagrant/Makefile as well

Done

**Comment by pmalek-sumo** [comment]
> newline before EOF

Should be good now 🤞🏻

**Review by sumo-drosiek — APPROVED** [review]

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Does the vagrant's `sumo-make yamllint` works fine?

**Review by sumo-drosiek — APPROVED** [review]

**Review by sumo-drosiek — COMMENTED** [review]
If you rebase over non-main PR, please set it as merge target for the review

**Pull request merged**